084-configure-env-vars-based-on-secrets-or-configmaps.md

```diff
@@ -0,0 +1,103 @@
+# Configure environment variables in all containers from Secrets or Config Maps
+
+This proposal introduces a new feature to allow configuring environment variables in any container deployed by Strimzi based on a value from a Secret or Config Map.
+
+## Current situation
+
+[Strimzi Proposal 75](https://github.com/strimzi/proposals/blob/main/075-additional-volumes-support.md) that was approved and implemented as part of Strimzi 0.43.0 gave Strimzi users the ability to mount custom volumes into any container deployed by Strimzi Cluster Operator.
+This includes mounting volumes based on Secrets and Config Maps.
+
+The [`ContainerEnvVar`](https://strimzi.io/docs/operators/latest/full/configuring.html#type-ContainerEnvVar-reference) API allows users to also specify environment variables for any container.
+But it currently allows users to set the environment variables only to plain text values directly inlined in the Strimzi custom resources.  
+The only exception to this are the Kafka Connect and MirrorMaker 2 containers, where additional environment variables based on Secret or Config Map can be added through the [external configuration](https://strimzi.io/docs/operators/0.43.0/full/configuring.html#type-ExternalConfiguration-reference) option.
+
+## Motivation
+
+The ability to specify environment variables for various containers based on Secret or Config Map comes up from time to time as a question or a feature request from our users.
+Often it is part of some advanced customization.
+For example passing credentials to various injected agents or passing decryption keys to files mounted from a Secret as a volume (for example when an encrypted certificate store is mounted as a volume and should be used).
+
+In addition, we now have an asymmetry in the Strimzi API design:
+* Users can now use Secrets and Config Maps as volumes in any container.
+* The [external configuration](https://strimzi.io/docs/operators/0.43.0/full/configuring.html#type-ExternalConfiguration-reference) part related to volumes has been deprecated and is planned to be removed in the future (with the v1 CRD API).
+* Environment variables in containers (except Kafka Connect and MirrorMaker 2 containers) can be defined only as plain values.
+* Environment variables based on Secrets and Config Maps can be defined for Kafka Connect and MirrorMaker 2 containers in the part of [external configuration](https://strimzi.io/docs/operators/0.43.0/full/configuring.html#type-ExternalConfiguration-reference) related to environment variables.
+  This part is currently not deprecated and not planned to be removed.
+
+## Proposal
+
+The [`ContainerEnvVar`](https://strimzi.io/docs/operators/latest/full/configuring.html#type-ContainerEnvVar-reference) will be extended to allow creating environment variables from a Secret or ConfigMap.
```

Comment:
```suggestion
The [`ContainerEnvVar`](https://strimzi.io/docs/operators/latest/full/configuring.html#type-ContainerEnvVar-reference) will be extended to allow creating environment variables from a Secret or ConfigMap. 
This extension will apply to all containers, including Kafka Connect and MirrorMaker 2, replacing the need to use the `ExternalConfiguration` schema for environment variables in these containers.
```

Reply:
I wonder if we need a line after this sentence to make it clear this is for all containers, including covering Kafka Connect and MirrorMaker 2 (rather than using the ExternalConfiguration schema).